### ISSUE TITLE
Debugging-related issues: 

### ISSUE THREAD

**Issue by eddytruyen** [FujitsuEnablingSoftwareTechnologyGmbH/kubernetes#22]
Thanks for sharing openstack-heat!

Using openstack-heat I have created a cluster with 2 minion nodes on a private openstack cloud.

I have 2 problems related to debugging:

1) `kubectl logs <Pod id>` gives as error that it can't find the host
```
[centos@kube-master-3 kube-yaml-files-of-services]$ kubectl get pods
NAME                      READY     STATUS              RESTARTS   AGE
mongo-controller-5pazv    0/1       ContainerCreating   0          13m
tomcat-controller-q1q7k   1/1       Running             0          1h
tomcat-controller-q7gj9   1/1       Running             0          1h
[centos@kube-master-3 kube-yaml-files-of-services]$ kubectl logs tomcat-controller-q1q7k
Error from server: Get https://kubernetesstack-node-e7rgqttj:10250/containerLogs/default/tomcat-controller-q1q7k/tomcat: dial tcp: lookup kubernetesstack-node-e7rgqttj on 8.8.8.8:53: no such host
```
2) On a minion node I can't find the logs of the kubelet agent. There is only a /var/log/es-kubelet.log.pos

Thanks for any advice

**Comment by cheld** [comment]
Your error message indicates that the host-name (kubernetesstack-node-e7rgqttj) is searched on Google DNS (8.8.8.8). A working DNS resolution for instance names is described as an openstack prerequisite in the documentation. http://kubernetes.io/docs/getting-started-guides/openstack-heat/

A private DNS can also be added as configuration value . This setting is applied to neutron. See: http://kubernetes.io/docs/getting-started-guides/openstack-heat/#set-additional-configuration-values

Out of my head, I am not sure how to create a cluster with only IPs.

Please note, we are not maintaining OpenStack heat any-more. You should post your issue on sig-openstack. We are focusing on docker-multinode and other approaches to simplify installation. 

We have created an experimental combination of docker-multinode (with enhancements) and heat. Depending on your requirements this might be a more easy approach towards a running cluster: 

https://github.com/FujitsuEnablingSoftwareTechnologyGmbH/k8s-docker-provisioner

(We are currently merging enhancements back to kube-deploy.)

**Comment by eddytruyen** [comment]
Interesting! I am also using docker-multinode on openstack. The project you mention  sounds very interesting.

Our research goals are performance evaluation of fully-features Kubernetes clusters on private openstack cloud 



Some of the features we'd like to evaluate:
* Automated migration of Cinder volumes: I've made for example a very simple extension to kube-deploy/docker-multinode that integrates with Flocker. The latter integrates with Cinder volumes. https://github.com/eddytruyen/kube-deploy
* I have also tried to run hyperkube with Cinder volumes. But here I run into the issue that the cinder plugin can't find attached devices by disk-id
https://github.com/kubernetes/kube-deploy/issues/182
* Services of type LoadBalancer (which should) automatically get a Loadbalancer via Neutron doesnt work

Have you been able to get  these features to work in https://github.com/FujitsuEnablingSoftwareTechnologyGmbH/k8s-docker-provisioner?

**Comment by cheld** [comment]
We have tested setting cloud provider and spend some considerable time debugging cinder volume. As far as I remember we did not fully solve the problem (basically same issue).

Running kubelet inside a container still has some issues, but it is getting close. As a quick solution, we bypassed the problem by extracting the kubelet binary to the host (which solves a couple of other issues as well). It would be easy to configure the cloud provider, but we have not implemented it.

We have had no chance to take a closer look on Flocker or OpenStack LoadBalancer. (But we contributed code to integrate OpenStack Monasca as Heapster sink)
